examples/search-knn.js

```diff
@@ -13,71 +13,79 @@ await client.connect();
 // Create an index...
 try {
   // Documentation: https://redis.io/docs/stack/search/reference/vectors/
-  await client.sendCommand([
-    "FT.CREATE",
-    "idx:knn-example",
-    "SCHEMA",
-    "v",
-    "VECTOR",
-    "HNSW",
-    "6",
-    "TYPE",
-    "FLOAT32",
-    "DIM",
-    "2",
-    "DISTANCE_METRIC",
-    "COSINE",
-  ]);
+  await client.ft.create('idx:knn-example', {
+    v: {
+      type: SchemaFieldTypes.VECTOR,
+      ALGORITHM: VectorAlgorithms.HNSW,
+      TYPE: 'FLOAT32',
+      DIM: 2,
+      DISTANCE_METRIC: 'COSINE'
+    }
+  }, {
+    ON: 'HASH',
+    PREFIX: 'noderedis:knn'
+  });
 } catch (e) {
-  if (e.message === "Index already exists") {
-    console.log("Index exists already, skipped creation.");
+  if (e.message === 'Index already exists') {
+    console.log('Index exists already, skipped creation.');
   } else {
     // Something went wrong, perhaps RediSearch isn't installed...
     console.error(e);
     process.exit(1);
   }
 }
 
-// This function accepts as input an array of numbers
-// and returns as output a byte representation that Redis accepts.
-const tobytes = (array) => Buffer.from(new Float32Array(array).buffer);
+function float32Buffer(arr) {
+  return Buffer.from(new Float32Array(arr).buffer);
+}
 
 // Add some sample data...
 // https://redis.io/commands/hset/
 await Promise.all([
-  client.hSet("noderedis:knn:a", { v: tobytes([0.1, 0.1]) }),
-  client.hSet("noderedis:knn:b", { v: tobytes([0.1, 0.2]) }),
-  client.hSet("noderedis:knn:c", { v: tobytes([0.1, 0.3]) }),
-  client.hSet("noderedis:knn:d", { v: tobytes([0.1, 0.4]) }),
+  client.hSet('noderedis:knn:a', { v: float32Buffer([0.1, 0.1]) }),
+  client.hSet('noderedis:knn:b', { v: float32Buffer([0.1, 0.2]) }),
+  client.hSet('noderedis:knn:c', { v: float32Buffer([0.1, 0.3]) }),
+  client.hSet('noderedis:knn:d', { v: float32Buffer([0.1, 0.4]) }),
 ]);
-
 // Perform a K-Nearest Neighbors vector similarity search
 // Documentation: https://redis.io/docs/stack/search/reference/vectors/#pure-knn-queries
-const results = await client.sendCommand([
-  "FT.SEARCH",
-  "idx:knn-example",
-  "*=>[KNN 4 @v $BLOB AS dist]",
-  "PARAMS",
-  "2",
-  "BLOB",
-  tobytes([0.1, 0.1]),
-  "SORTBY",
-  "dist",
-  "DIALECT",
-  "2",
-]);
-
+const results = await client.ft.search('idx:knn-example', '*=>[KNN 4 @v $BLOB AS dist]', {
```

Comment:
AMAZING! I could not for the life of me figure out how to formulate this with the `ft.search` API. This is perfect.